### PR TITLE
Fix responsive spacing issues for show map button

### PR DESCRIPTION
**Pull request by kccrs**
# Purpose 

On smaller screen sizes the show map button was resizing too small and did not have enough breathing room between its neighbor.

I also updated many elements including the header as it had some responsive issues of its own. There were many challenges in that many of the elements have the same classname and were styled the same even though the layout was different between the admin and user pages.  I added additional classnames to some of the elements in order to ensure layout differences remained intact. 

I had to manually copy some code from master into my branch that didn't come over when I pulled down for some reason, so that is why some of the changes appear in Admin and App components.  I did not want to lose the already approved and merged changes.

### CODE REVIEW

File: lib/components/App.js [hunk 1]
```diff
@@ -184,46 +183,37 @@ export default class App extends Component {
...
-    }
-    else if (speaker.attendees && attendingSpeaker) {
-      speaker.attendees = speaker.attendees.filter(attendee => attendee !== user.email);
-    }
-    else {
-      speaker.attendees = [user.email];
-    }
-    firebase.database().ref(`speakers/${speaker.key}`).update( speaker );
-  }
-
```
Comment (swanie21) [inline lib/components/App.js]: Do we not need this function anymore for gear ups or the empty array as the state for gearUp? I still have it in my master.

File: lib/components/Header.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
 
 const Header = ({ admins, user, createSpike, showForm, toggleForm }) => {
   return (
-    <header className="Header">
+    <header>
```
Comment (swanie21) [inline lib/components/Header.js]: Good catch to remove the redundancy.

File: lib/components/App.js [hunk 3]
```diff
@@ -236,29 +226,18 @@ export default class App extends Component {
...
+            attending={attending}
             createSpike={(e) => this.createSpike(e)}
-            spikes={this.state.spikes}
+            spikes={spikes}
             user={user}
-            updateSpike={(spike, prop, value) => this.updateSpike(spike, prop, value)}
             updateCount={(spike) => this.updateCount(spike)}
-            assignRooms={this.assignRooms}
-            admins={this.state.admins}
-            deleteSpike={(spike) => this.deleteSpike(spike)}
```
Comment (swanie21) [inline lib/components/App.js]: Are we not passing a lot of these props to components anymore?

**Comment by swanie21** [comment]
It looks like the show map button was fixed, which was the main purpose of this PR and a lot of other changes were made. On mobile the button fonts on these buttons: Admins List, Show Map, Add Guest Speaker, Add Gear Up, Sign Out and Add Spike get smaller, which I think it should be the same size as desktop or even larger since it's on a smaller device, so users can see the text easier. Also the sign out button is briefly at the top of the profile pic before it pops down below the pic at a media query. The header responsive design definitely looks better though. I think the sign out button shouldn't be popping around though and with the media query that aligns the profile pic in the middle it should just go beneath it at that point. 

It looks like there was a lot of JS changes on the App component. Can you help explain why these changes were made? 

Otherwise the responsive design looks much better!

File: lib/components/App.js (hunk 1 above)
Comment (Jeff-Duke) [inline lib/components/App.js]: We definitely need this still.  This is essential for the RSVP functionality on the guest speaker.

**Comment by Jeff-Duke** [comment]
This PR breaks the guest speaker RSVP functionality and gets rid of the gear-up section.  We need to make sure we have existing functionality before merging this PR.

File: lib/components/App.js [hunk 4]
```diff
@@ -184,46 +183,37 @@ export default class App extends Component {
...
             newAdmin={(email) => this.newAdmin(email)}
             removeAdmin={(index) => this.removeAdmin(index)}
-            user={this.state.user}
-            speaker={this.state.speaker}
-            gearUp={this.state.gearUp}
-            assignRooms={this.assignRooms}
-            sanitizeInput={this.sanitizeInput}
+            user={user}
+            speaker={speaker}
           />
```
Comment (Jeff-Duke) [inline lib/components/App.js]: We definitely still need to pass these functions down.  I see that you're deconstructing state to make things cleaner but we still need the methods.

File: lib/components/App.js (hunk 3 above)
Comment (Jeff-Duke) [inline lib/components/App.js]: Same as above, I like deconstructing state to pass down props, we need to make sure we're still passing down methods like updateSpeakerAttendees, deleteSpike, updateSpike etc.

File: lib/components/App.js (hunk 1 above)
Comment (kccrs) [inline lib/components/App.js]: Hmm... for some reason when I pulled down master, it didn't bring all the changes with it.... I copied and pasted stuff that I thought was missing, but I guess I missed a few things.

Comment (kccrs) [inline lib/components/App.js]: Fixed.

File: lib/components/App.js (hunk 3 above)
Comment (kccrs) [inline lib/components/App.js]: Fixed.

File: lib/components/App.js (hunk 4 above)
Comment (kccrs) [inline lib/components/App.js]: Yep, you're right... I lost those in the merge.  Fixed.

File: lib/components/Header.js (hunk 2 above)
Comment (kccrs) [inline lib/components/Header.js]: yeah we weren't using it in our Sass. Since there's only one header, we didn't need a class.  :)

**Comment by kccrs** [comment]
Ok, I think I have fixed all the issues in this PR and updated a few other things that I noticed... you know.. change one thing, something else breaks.  Pull it down, take a look and make sure everything looks OK and is working correctly.

**Comment by Jeff-Duke** [comment]
As far as I can tell this all looks good now.  Can someone else pull down the branch and confirm?

**Pull request merged**